mi/dataset/driver/zplsc_c/zplsc_echogram_generator.py

```diff
@@ -0,0 +1,633 @@
+        passed in.
+
+        :param subsite: The subsite of the ZPLSC instrument.
+        :return: deployments: The list of deployment directories.
+        """
+
+        deployments = [deployment for deployment in self.deployments]
+        if not deployments:
+            # Generate the portion of the URL up to the subsite directory.
+            subsite_url = join(RAW_DATA_URL, subsite.upper())
+            try:
+                # Get the all the deployment sub-directories under the subsite directory.
+                deployment_dirs = self.get_dir_contents(subsite_url)
+            except urllib2.HTTPError:
+                raise
+
+            # Generate the list of the 24 1-hour raw data files for the given date.
+            deployment_list = DEPLOYMENT_DIR_MATCHER.findall(deployment_dirs)
+            deployment_list.sort()
+
+            if self.process_mode:
+                deployment_list = [deployment_list[-1]]
+
+            for deployment in deployment_list:
+                try:
+                    deployments.append(int(deployment))
+
+                except ValueError as ex:
+                    log.error('Invalid deployment number: %s: %s', deployment, ex.message)
+                    break
+
+        return deployments
+
+    def store_echogram(self, zplsc_echogram_file):
+        """
+        This method will store the generated echogram in the location specified
+        in the configuration file. It will then clean up the temporary folder.
+
+        :param zplsc_echogram_file: The generated ZPLSC echogram
+        :return:
+        """
+
+        # Create the echogram directory, then copy the echogram to the local directory.
+        self.echogram_directory = expanduser(self.echogram_directory)
+        if not exists(self.echogram_directory):
+            mkdir(self.echogram_directory)
+
+        system('cp ' + zplsc_echogram_file + ' ' + self.echogram_directory)
+
+        # Clean up the temporary files.
+        system('rm -f ' + join(self.temp_directory, '*'))
+
+    def generate_zplsc_echograms(self):
+        """
+        This method will get the subsites, deployments and dates from the
+        command line or all of the subsites, deployments and dates for the
+        daily process.  It will generate the echograms based on those inputs
+        and upload the echograms to the raw data server.
+
+        :return:
+        """
+
+        # Create the temporary data file directory.
+        self.temp_directory = join(expanduser(USER_HOME), TEMP_DIR)
+        if not exists(self.temp_directory):
+            mkdir(self.temp_directory)
+
+        # Create the echograms for the zplsc instruments of each subsite.
+        for subsite in self.subsites:
+            zplsc_24_subsite_prefix = subsite + '-'
+
+            try:
+                deployments = self.get_deployment_dirs(subsite)
+            except urllib2.HTTPError:
+                continue
```

Comment:
Is this worth a warning? If the subsite's deployment directories are missing, isn't that unexpected?

Reply:
See the comment above.